lib/SPIRVProducerPass.cpp

```diff
@@ -496,8 +427,40 @@ struct SPIRVProducerPass final : public ModulePass {
   // Returns true if |Arg| is called with a coherent resource.
   bool CalledWithCoherentResource(Argument &Arg);
 
+  //
+  // Primary interface for adding SPIRVInstructions to a SPIRVSection.
+  template <enum SPIRVSection TSection = kFunctions>
```

Comment:
Why templating over just taking an extra parameter? Seems like a waste for the compiler to generate extra functions to start with. I'd also rather be explicit everywhere than just making kFunction instructions easier to insert.

Reply:
Since every call uses an explicit Section the compiler can optimize for it and should be pretty insignificant for compile time since there are only 11 specializations.